File: other_problems/MinimumSizeSubarraySum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int sz = nums.size();
        int minimal =  sz + 1;
        int sum = 0;

        int i = 0, j = 0;
        while(j <= sz ){
            if(sum >= target){
                minimal = min(minimal, j - i);
                sum -= nums[i++];
            }
            else{
                if(j == sz)
                    j++;
                else
                    sum += nums[j++];
            }
        }

        return (minimal == sz + 1? 0 : minimal);
    }
};

File: phase_one_eduacation/week4/leetcode/minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        left = 0
        window = 0
        res = len(nums) + 1
        for right in range(len(nums)):
            window += nums[right]
            while window >= target:
                res = min(res, right - left + 1)
                window -= nums[left]
                left += 1
           
        if res > len(nums):
            return 0
        return res